### DISCUSSION TITLE
A new era for Angular - Zoneless Change Detection - justangular.com

### DISCUSSION THREAD

**Discussion by eneajaho** [eneajaho/just-angular#15] [General]
https://justangular.com/blog/a-new-era-for-angular-zoneless-change-detection

**Comment by AmirAlOmari** [comment]
Great article, thanks for sharing!
Though I believe it would be great to hear what exactly can go _wrong_ using this experimental API and how to avoid it.

**Reply by eneajaho** [reply]
The only thing that can go wrong is your app won't work as it's supposed too, this can be caused by components that don't work correctly on zoneless or third party libs which don't support zoneless yet.

**Comment by adelamata** [comment]
Hi, great article I really like it. I have a question about of zone removing, This remove has a real impact on the applications efficiency or the idea only is deleting external dependencies?. 

Thank you!!

**Reply by eneajaho** [reply]
Hi, this would impact also the performance. Loading zone.js in application startup takes a little amount of time, and that won't be the case anymore. Also, debugging will become 100x better, because zone.js won't be there anymore and you can read easily the code that is being run. Also, bundle size gets better because we remove zone.js (around 13kb gziped).

**Reply by adelamata** [reply]
Hello @eneajaho, perfect. Thank you very much!

**Comment by laurent67100** [comment]
Thank you for this great explanation!

**Comment by sdedieu** [comment]
During the Signal RFC discussion, I remembered it was mentioned that signal components would redraw only the affected DOM View, not the whole component DOM. Are we agree that this is not done yet ? 🙂
[Playground](https://stackblitz.com/edit/stackblitz-starters-udmu3z?file=src%2Fmain.ts)

**Reply by eneajaho** [reply]
Yeah, signal components are not released yet. What we have now are just signal API-s but not signal components.

**Comment by nomasfabian** [comment]
I just implemented zoneless in my angular18 app, but im having some troubles with some components, for example bootstrap tooltips, it never displays the tooltip, i guess not every library is prepared for this. 

so you may get yourself in some trouble looking for control libraries for your app, that dont use the detection change strategy onPush

**Comment by TheAjinkya** [comment]
Thanks for making such a deep and thorough blog with all the concepts, hatts off!
